design/parallel-commit/globally-non-unique-timestamps.md

```diff
@@ -22,6 +22,11 @@ In fact it would be simpler to simply disallow the locking in the above case, ju
 
 (Note: Now on a pessimistic lock, if the commit_ts of the existing commit record are the same as the current for_update_ts, then it is allowed to lock successfully)
 
+### commit-commit conflict
+
+It is possible two transactions have the same `commit_ts`. It's easy to imagine one transaction gets a `commit_ts` as `max(max_read_ts) + 1` and another gets that timestamp from PD. This is fine as long as the two transactions don't meet. If the two transactions try to write the same key, then there would be two competing values for any reads after that `commit_ts` (TiKV could not write both values into the write CF, but this is a technicality, the more fundamental problem is that there is no way for TiKV to judge which value is most recent).
```

Comment:
I'm not sure, when is this situation that two txn commits the same key with the same commit_ts possible? Isn't it that on a same key, a transaction should have either start_ts or for_update_ts not less than previously written transaction's commit_ts to be successful?

Reply:
Ah, I cannot think of a case when two commits can conflict, either. Rollback seems the only case when two write cf records conflict.

Reply:
I think it is possible that two transactions have the same commit ts, but not that they can both affect the same key due to locking the key.